solutions/azure-hub-spoke-connected-group/bicep/modules/avnm.bicep

```diff
@@ -0,0 +1,315 @@
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(resourceGroup().id, userAssignedIdentity.name)
+  properties: {
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c') // Contributor: b24988ac-6180-42a0-ab88-20f7382dd24c
+    principalId: userAssignedIdentity.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+@description('This is the securityadmin configuration assigned to the AVNM')
+resource securityConfig 'Microsoft.Network/networkManagers/securityAdminConfigurations@2022-05-01' = {
+  name: 'sg-${location}'
+  parent: networkManager
+   properties: {
+    applyOnNetworkIntentPolicyBasedServices: ['None'] 
+    description: 'Security Group for AVNM'
+     }
+  }
+
+@description('This is the rules collection for the security admin config assigned to the AVNM')
+resource rulesCollection 'Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections@2022-05-01' = {
+name: 'rc-${location}'
+parent: securityConfig
+properties: {
+  appliesToGroups: [
+     {
+      networkGroupId: networkManager::networkGroupAll.id
+     }
+   ]
+ }
+}
+
+@description('This is the rule is to Deny all TCP Rules')
+resource rule1 'Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/rules@2022-05-01' = if (deployDefaultDenySecurityAdminRules) {
+name: 'r-tcp-${location}'
+kind: 'Custom'
+parent: rulesCollection
+properties: {
+  access: 'Deny'
+  description: 'Denying TCP rules'
+  destinationPortRanges: ['20','21','22','23','69','119','161','445','512','514','873','3389','5800','5900']
+  destinations: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]
+  direction: 'Inbound'
+  priority: 100
+  protocol: 'TCP'
+  sourcePortRanges: ['0-65535']
+  sources: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]  
+  }      
+}
+
+@description('This rule to deny inbound TCP/UDP rules')
+resource rule2 'Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/rules@2022-05-01' = {
+name: 'r-tcp-udp-${location}'
+kind: 'Custom'
+parent: rulesCollection
+properties: {
+  access: 'Deny'
+  description: 'Deny all TCP/UDP rules'
+  destinationPortRanges: ['11','135','162','593','2049']
+  destinations: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]
+  direction: 'Inbound'
+  priority: 101
+  protocol: 'TCP,UDP'
+  sourcePortRanges: ['0-65535']
+  sources: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]  
+  }      
+}
+
+@description('This rule to deny inbound UDP rules')
+resource rule3 'Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/rules@2022-05-01' = {
```

Comment:
Where are the NSG rule or their analogs from the tranditional hub-spoke example?  For example: 
```
securityRules: [
      {
        name: 'AllowAll443InFromVnet'
        properties: {
          protocol: 'Tcp'
          sourcePortRange: '*'
          sourceAddressPrefix: 'VirtualNetwork'
          destinationPortRange: '443'
          destinationAddressPrefix: 'VirtualNetwork'
          access: 'Allow'
          priority: 100
          direction: 'Inbound'
        }
      }
      {
        name: 'DenyAllInbound'
        properties: {
          protocol: '*'
          sourcePortRange: '*'
          sourceAddressPrefix: '*'
          destinationPortRange: '*'
          destinationAddressPrefix: '*'
          access: 'Deny'
          priority: 1000
          direction: 'Inbound'
        }
      }
      {
        name: 'DenyAllOutbound'
        properties: {
          protocol: '*'
          sourcePortRange: '*'
          sourceAddressPrefix: '*'
          destinationPortRange: '*'
          destinationAddressPrefix: '*'
          access: 'Deny'
          priority: 1000
          direction: 'Outbound'
        }
      }
   ```   

I feel like we should make sure we are trying to show a compatible story.  Am I missing them in here somewhere?

Reply:
We didn't also include NSGs for simplicity, but I think it is a valid point that showing the NSG/AVNM interaction would have value. I'll add some basic NSGs, with a deny outbound rule that the always allow admin rule will override

Reply:
Just match the NSG rules in the existing hub-spoke please.  No need to invent something new.